maintainers/scripts/haskell/sweep-overrides.sh

```diff
@@ -0,0 +1,70 @@
+set -euo pipefail
+
+function help {
+    echo "Usage: $0 PROPERTY [NIXPKGS] [ATTRIBUTE]"
+    echo
+    echo "Search for unnecessary property overrides by attempting builds with the default value"
+    echo
+    echo "  PROPERTY                  one of {jailbreak, doCheck, doHaddock}"
+    echo "  NIXPKGS                   file path to nixpkgs"
+    echo "  PACKAGE_SET               nix function returning an haskell package set"
+    echo
+    echo "Example:"
+    echo "  \$ $0 jailbreak path/to/nixpkgs '(pkgs: pkgs.haskellPackages)'"
+
+    exit 1
+}
+
+if [[ $# < 1 ]]; then
+    help
+fi
+
+property=$1
+shift
+
+nix_args=""
+
+if [[ $# > 0 ]]; then
+    nix_args+=" nixpkgsPath = $1;"
+    echo $nix_args
+    shift
+fi
+
+if [[ $# > 0 ]]; then
+    nix_args+=" getPackageSet = $1;"
+    echo $nix_args
+    shift
+fi
+
+if [[ $# > 0 ]]; then
+    help
+fi
+
+dir=$( cd -- "$( dirname -- "${BASH_SOURCE[0]}" )" &> /dev/null && pwd )
+
+expression="with (import $dir/sweep-overrides.nix {$nix_args}); modified lenses.$property"
+
+#TODO: filter out packages to avoid unbuildable ones making their way into the closure
+export NIXPKGS_ALLOW_BROKEN=1
+export NIXPKGS_ALLOW_UNSUPPORTED_SYSTEM=1
```

Comment:
I'm not sure how to simply filter these out given they are transitive.

Reply:
`(builtins.tryEval drv.outPath).success`.

Reply:
Worked